ex/app/src/main/java/com/example/ex/ScrollingActivity.java

```diff
@@ -0,0 +1,68 @@
+package com.example.ex;
+
+import android.os.Bundle;
+import com.google.android.material.appbar.CollapsingToolbarLayout;
+import androidx.appcompat.app.ActionBar;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import android.util.Log;
+import android.widget.RatingBar;
+
+public class ScrollingActivity extends AppCompatActivity implements IAttachable {
+
+    private State state;
```

Comment:
We shouldn't hold state like that. `Activities` and `Fragment`s are views and should be dummy. They only render what `ViewModel` tells them to. Let's just observe some `LiveData` in our `Activity` and render the data instead. And `ViewModel` will be notified about the user interactions via some `onWhateverChanged` methods